sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DeploymentStep.java

```diff
@@ -176,25 +177,48 @@ private void setTaskStatus(Protos.TaskID taskID, Status status) {
         }
     }
 
-    private Status getStatus(Map<Protos.TaskID, TaskStatusPair> tasks) {
+    @VisibleForTesting
+    Status getStatus(Map<Protos.TaskID, TaskStatusPair> tasks) {
         if (tasks.isEmpty()) {
             return Status.PENDING;
         }
 
+        Set<Status> statuses = new HashSet<>();
         for (Map.Entry<Protos.TaskID, TaskStatusPair> entry : tasks.entrySet()) {
             String taskId = entry.getKey().getValue();
             Status status = entry.getValue().getStatus();
             logger.info("TaskId: {} has status: {}", taskId, status);
-            if (!status.equals(Status.COMPLETE)) {
-                // Keep and log current status
-                return super.getStatus();
-            }
+
+            statuses.add(status);
+        }
+
+        // A DeploymentStep should have the "least" status of its consituent tasks.
+        // We can rely on the order of Status as returned by its getOrder method.
+        // For example:
+        // 1 PENDING, 2 STARTING => PENDING
+        // 2 STARTING, 1 COMPLETE => STARTING
+        // 3 COMPLETE => COMPLETE
```

Comment:
I could read this, but someone new to the code might not be able to quickly understand that those numbers mean counts of Tasks.

Reply:
Mentioning the `getOrder` method is no longer applicable, right?